src/gravel/controllers/containers.py

```diff
@@ -26,6 +27,60 @@ class ContainerPullError(ContainerError):
     pass
 
 
+class UnknownRegistryError(ContainerError):
+    pass
+
+
+class RegistrySecurityError(ContainerError):
+    pass
+
+
+class ImageNameError(ContainerError):
+    pass
+
+
+class UnknownImageError(ContainerError):
+    pass
+
+
+class UnknownImageTagError(ContainerError):
+    pass
+
+
+async def registry_check(registry: str, image: str, secure: bool) -> None:
+
+    proto = "https" if secure else "http"
+    # check for an existing registry
+    #
+    try:
+        req = requests.get(f"{proto}://{registry}/v2/", timeout=5)
+        if req.status_code != requests.codes.ok:
+            raise UnknownRegistryError(msg=registry)
+    except requests.exceptions.SSLError:
+        raise RegistrySecurityError(msg=registry)
+    except requests.exceptions.ConnectionError:
+        raise UnknownRegistryError(msg=registry)
+
+    # check for image:tag
+    #
+    img = image.split(":")
+    if len(img) != 2:
+        raise ImageNameError(msg=image)
+    (imgname, imgtag) = img
+
+    try:
+        req = requests.get(f"{proto}://{registry}/v2/{imgname}/tags/list")
+    except Exception as e:
+        raise ContainerError(msg=str(e))
+
+    if req.status_code != requests.codes.ok:
+        raise UnknownImageError(msg=imgname)
```

Comment:
I prefer messages that are a little more meaningful. That's because if such an exception is processed by an API endpoint which simply converts it as a HTTPException with message `str(err)` this message is not really helpful to the user when it is shown in the UI.
```suggestion
        raise UnknownImageError(msg=f"The image '{imgname}' does not exist.")
```

Reply:
My only counter argument here is that the exception itself is already pretty self-explanatory, and the contents of the exception message should just clarify it. IMO, it's on the caller's plate to translate that to something that can be wired through to the API consumer.

Granted that's not happening at the moment, but it's beyond the scope of this PR.